### PR TITLE
Terminal account page

### PR DESCRIPTION
**Pull request by Sirmorrison**
added account page to terminal ui.
Fixes issues #374

image

image

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -265,9 +265,9 @@ func enterFyneMode(ctx context.Context, walletMiddleware app.WalletMiddleware) {
 	beginShutdown <- true
 }
 
-func enterTerminalMode(ctx context.Context, walletMiddleware app.WalletMiddleware) {
+func enterTerminalMode(ctx context.Context, walletMiddleware app.WalletMiddleware, appConfig *config.Config) {
```
Comment (ademuanthony) [inline main.go]: I think you only need `*config.Settings`, this function can accept just that and have it passed down below

**Review by ademuanthony — CHANGES_REQUESTED** [review]
Unable to view account details
image

File: main.go (hunk 1 above)
Comment (Sirmorrison) [reply main.go]: It gives a type error. `cannot use Application (type *config.Config as type *config.Settings in argument to enterTerminalMode`

**Comment by Sirmorrison** [comment]
> Unable to view account details
> image

Resolved.

File: main.go [hunk 2]
```diff
@@ -120,7 +120,7 @@ func main() {
 	case "fyne":
 		enterFyneMode(ctx, walletMiddleware)
 	case "terminal":
-		enterTerminalMode(ctx, walletMiddleware)
+		enterTerminalMode(ctx, walletMiddleware, appConfig)
```
Comment (ademuanthony) [inline main.go]: you should pass `appConfig.Settings` here and change the parament for the `enterTerminalMode` to accept `config.Settings` as suggested in an earlier comment

**Review by ademuanthony — CHANGES_REQUESTED** [review]

File: main.go [hunk 3]
```diff
@@ -265,9 +265,9 @@ func enterFyneMode(ctx context.Context, walletMiddleware app.WalletMiddleware) {
 	beginShutdown <- true
 }
 
-func enterTerminalMode(ctx context.Context, walletMiddleware app.WalletMiddleware) {
+func enterTerminalMode(ctx context.Context, walletMiddleware app.WalletMiddleware, appConfig config.Settings) {
```
Comment (ademuanthony) [inline main.go]: `appConfig` should be `settings` or `appSettings`

**Review by ademuanthony — CHANGES_REQUESTED** [review]

File: terminal/pages/accounts.go [hunk 4]
```diff
@@ -1,28 +1,251 @@
...
+		tviewApp.SetFocus(hideAccount)
+		displayAccountsDetails(wallet.NetType(), selectedAccount, accountPropertiesTable, displayMessage)
+
+		if settings.DefaultAccount != selectedAccount.Number {
+			defaultAccount.SetChecked(false)
+		} else {
+			defaultAccount.SetChecked(true)
+		}
+
+		var accountExist bool
```
Comment (ademuanthony) [inline terminal/pages/accounts.go]: this variable can be renamed to `accountIsHidden` I guess

**Review by ademuanthony — CHANGES_REQUESTED** [review]

**Review by ademuanthony — APPROVED** [review]

**Pull request merged**